pkg/workspace/apply.go

```diff
@@ -42,15 +45,47 @@ func (nvm nameVolumeMap) setVolumeSource(workspaceName string, volumeName string
 	}
 }
 
+// GenerateHashedName generates a unique name for a volume based on the workspace name.
```

Comment:
NIT: there is nothing specific to volumes or workspaces in this function

Reply:
NIT: this could have lived in the `pkg/names` even though the interface is slightly different. Using this with a different prefix and length might get names that are too long for k8s. Of course it is not going to happen with the code as it is today, but it might be worth adopting an approach similar to that in `pkg/names` where the overall length is restricted.